File: dlna-dms/src/main/java/com/android/cast/dlna/dms/ContentFactory.java
package com.android.cast.dlna.dms;

import android.content.Context;

import androidx.annotation.Nullable;

import org.fourthline.cling.support.contentdirectory.ContentDirectoryException;
import org.fourthline.cling.support.model.BrowseResult;

public class ContentFactory {

    private static class Holder {
        private static final ContentFactory sInstance = new ContentFactory();
    }

    public static ContentFactory getInstance() {
        return Holder.sInstance;
    }

    private ContentFactory() {
    }

    private IContentFactory mContentFactory;

    public void setServerUrl(Context context, String url) {
        mContentFactory = new IContentFactory.ContentFactoryImpl(context, url);
    }

    @Nullable
    public BrowseResult getContent(String objectID) throws ContentDirectoryException {
        if (mContentFactory != null) {
            return mContentFactory.getBrowseResult(objectID);
        }
        return null;
    }
}

File: dlna-dms/src/main/java/com/android/cast/dlna/dms/ContentFactory.kt
package com.android.cast.dlna.dms

import android.content.Context
import com.android.cast.dlna.dms.IContentFactory.ContentFactoryImpl
import org.fourthline.cling.support.contentdirectory.ContentDirectoryException
import org.fourthline.cling.support.model.BrowseResult

object ContentFactory {

    private var contentFactory: IContentFactory? = null

    fun setServerUrl(context: Context?, url: String?) {
        contentFactory = ContentFactoryImpl(context, url)
    }

    @Throws(ContentDirectoryException::class)
    fun getContent(objectID: String?): BrowseResult? = contentFactory?.getBrowseResult(objectID)
}